frontend/packages/process-editor/src/hooks/useBpmnEditor.ts

```diff
@@ -7,53 +7,56 @@ import type { TaskEvent } from '../types/TaskEvent';
 import type { SelectionChangedEvent } from '../types/SelectionChangeEvent';
 import { getBpmnEditorDetailsFromBusinessObject } from '../utils/bpmnObjectBuilders';
 import { useStudioRecommendedNextActionContext } from '@studio/components';
+import type Modeler from 'bpmn-js/lib/Modeler';
 
 // Wrapper around bpmn-js to Reactify it
 
 export type UseBpmnEditorResult = (div: HTMLDivElement) => void;
 
 export const useBpmnEditor = (): UseBpmnEditorResult => {
-  const { getUpdatedXml, bpmnXml, modelerRef, setBpmnDetails } = useBpmnContext();
+  const { getUpdatedXml, setBpmnDetails } = useBpmnContext();
   const { metadataFormRef, resetForm } = useBpmnConfigPanelFormContext();
   const { addAction } = useStudioRecommendedNextActionContext();
 
   const { saveBpmn, onProcessTaskAdd, onProcessTaskRemove } = useBpmnApiContext();
 
-  const handleCommandStackChanged = async () => {
-    saveBpmn(await getUpdatedXml(), metadataFormRef.current || null);
+  const handleCommandStackChanged = useCallback(async () => {
+    const xml = await getUpdatedXml();
+    saveBpmn(xml, metadataFormRef.current || null);
     resetForm();
-  };
-
-  const handleShapeAdd = (taskEvent: TaskEvent): void => {
-    const bpmnDetails = getBpmnEditorDetailsFromBusinessObject(taskEvent?.element?.businessObject);
-    onProcessTaskAdd({
-      taskEvent,
-      taskType: bpmnDetails.taskType,
-    });
-    if (
-      bpmnDetails.taskType === 'data' ||
-      bpmnDetails.taskType === 'payment' ||
-      bpmnDetails.taskType === 'signing'
-    )
-      addAction(bpmnDetails.id);
-  };
-
-  const handleShapeRemove = (taskEvent: TaskEvent): void => {
-    const bpmnDetails = getBpmnEditorDetailsFromBusinessObject(taskEvent?.element?.businessObject);
-    onProcessTaskRemove({
-      taskEvent,
-      taskType: bpmnDetails.taskType,
-    });
-  };
-
-  const handleSelectionChange = (selectionEvent: SelectionChangedEvent): void => {
-    if (selectionEvent.newSelection.length !== 1) {
-      setBpmnDetails(null);
-      return;
-    }
-    const selectedElement = selectionEvent.newSelection[0];
-    updateBpmnDetails(selectedElement);
-  };
+  }, [saveBpmn, resetForm, metadataFormRef, getUpdatedXml]);
+
+  const handleShapeAdd = useCallback(
+    async (taskEvent: TaskEvent): Promise<void> => {
+      const bpmnDetails = getBpmnEditorDetailsFromBusinessObject(
+        taskEvent?.element?.businessObject,
+      );
+      onProcessTaskAdd({
+        taskEvent,
+        taskType: bpmnDetails.taskType,
+      });
+      if (
+        bpmnDetails.taskType === 'data' ||
+        bpmnDetails.taskType === 'payment' ||
+        bpmnDetails.taskType === 'signing'
```

Comment:
Should we move these three in to a variable? 😄

Reply:
Would it make more sense having this call to `addAction` as a function executed inside `onProcessTaskAdd`? 
No need to fix now, but maybe something we can consider in a future cleanup.

Reply:
This is one of many things that could be improved in this file. An even better solution would be to create an independent function in a separate file. But I would need more time in order to evaluate what the `addAction` function is actually doing and find proper names. So to keep the pull request concise, I have restricted changes inside the event listener functions to a minimum.

Reply:
> Would it make more sense having this call to addAction as a function executed inside onProcessTaskAdd?
That's also a possible solution. There are many solutions to consider here, so that's why I didn't want to touch it this time.